### ISSUE TITLE
configurable preference for nameservers with forward plugin

### ISSUE THREAD

**Issue by yrro** [miekg/coredns#6305]
**What would you like to be added**: a way to specify a preference value for upstream nameservers in the configuration of the `forward` plugin

**Why is this needed**:

Let's say I have a network spanning several sites, and I'm using coredns to provide recursive DNS at each site.

Each site has its own authoritative nameservers which host private DNS zones for use within the network.

I'd like to be able to configure coredns at a site to use the site's own authoritative nameservers, and only query the nameservers of other site's if this site's nameservers are not responding.

I believe the best I can do at the moment is something like:

```
int.example.com {
  forward . 192.0.2.53 192.0.2.54 198.51.100.53 203.0.113.53 198.51.100.54 203.0.113.54 {
    policy sequential
  }
}
```

The first two nameservers are those at the local site (192.0.2.0/24). The other four nameservers are at the remote sites (198.51.100.0/24 and 203.0.113.0/24).

As a result, query load is not spread equally between the two nameservers at the local site, rather all queries go to the first nameserver unless it's down.

Only if both local nameservers are down to queries to go remote sites (good) but all such queries will go to the first nameserver in site 2, unless it's down, in which case they go to the first nameserver in site 3, and so on.

I'm going to invent a syntax here to show what I'd like. The exact syntax doesn't matter, so don't worry about that, this is just to show what I'd like to be able to do when configuring coredns:

```
int.example.com {
  forward . 192.0.2.53;pref=0 192.0.2.54;pref=0 198.51.100.53;pref=10 203.0.113.53;pref=20 198.51.100.54;pref=10 203.0.113.54;pref=20 {
    policy random
  }
}
```

The behaviour would be: queries are distributed randomly between the nameservers with preference 0. Only if both nameservers are not responding are queries then distributed randomly between the nameservers with preference 10. If _those_ are both down then queries are distributed randomly between the nameservers with preference 20.

To summarize, nameservers are grouped by preference value. Then the configured forward policy is applied between nameservers with the same preference value. Nameservers with higher preference are only tried if all the nameservers with lower preference value are down.

**Comment by SuperQ** [comment]
What if instead we made it possible for `forward` to continue?

Something like this:
```
int.example.com {
  # pref=0
  forward . 192.0.2.53 192.0.2.54 {
    policy random
    continue true
  }
  # pref=10
  forward . 198.51.100.53 198.51.100.54 {
    policy random
    continue true
  }
  # pref=20
  forward . 203.0.113.53 203.0.113.54 {
    policy random
    continue false
  }
}
```

**Comment by yrro** [comment]
That would work for me!
